network/src/protocols/rpc/mod.rs

```diff
@@ -297,12 +297,15 @@ async fn handle_outbound_rpc<TSubstream>(
 
             // Future to run the actual outbound rpc protocol and get the results.
             let mut f_rpc_res = handle_outbound_rpc_inner(peer_mgr_tx, peer_id, protocol, req_data)
-                .boxed()
-                .compat()
                 .timeout(timeout)
-                .compat()
-                // Convert tokio timeout::Error to RpcError
-                .map_err(Into::<RpcError>::into);
+                .map_err(Into::<RpcError>::into)
+                .map(|r| match r {
+                    Ok(Ok(x)) => Ok(x),
+                    Ok(Err(e)) => Err(e),
+                    Err(e) => Err(e),
+                })
```

Comment:
Is this due to changes in upstream timeout future? Also, should we create a function to collapse a `Result<Result<T, E>, E>` into a `Result<T, E>` and use that at various timeout callsites?